examples/fabric/byot/trainer.py

```diff
@@ -0,0 +1,409 @@
+import os
+from collections.abc import Mapping
+from functools import partial
+from typing import Any, cast, List, Literal, Optional, Tuple, Union
+
+import torch
+from lightning_utilities.core import is_overridden
+from lightning_utilities.core.apply_func import apply_to_collection
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+
+from lightning.fabric.fabric import (
+    _PLUGIN_INPUT,
+    _PRECISION_INPUT,
+    _unwrap_objects,
+    Accelerator,
+    Fabric,
+    Logger,
+    Strategy,
+)
+from lightning.fabric.strategies.fsdp import FSDPStrategy
+from lightning.fabric.utilities.rank_zero import rank_zero_warn
+from lightning.fabric.utilities.types import LRScheduler, Optimizable
+from lightning.pytorch.core.module import LightningModule
+
+
+class Trainer:
+    def __init__(
+        self,
+        accelerator: Union[str, Accelerator] = "auto",
+        strategy: Union[str, Strategy] = "auto",
+        devices: Union[List[int], str, int] = "auto",
+        precision: _PRECISION_INPUT = "32-true",
+        plugins: Optional[Union[_PLUGIN_INPUT, List[_PLUGIN_INPUT]]] = None,
+        callbacks: Optional[Union[List[Any], Any]] = None,
+        loggers: Optional[Union[Logger, List[Logger]]] = None,
+        max_epochs: Optional[int] = 1000,
+        max_steps: Optional[int] = None,
+        grad_accum_steps: int = 1,
+        limit_train_batches: Union[int, float] = float("inf"),
+        limit_val_batches: Union[int, float] = float("inf"),
+        val_sanity_check: int = 2,
+        validation_frequency: int = 1,
```

Comment:
did you intentionally rename these?

Reply:
Personally, I found these names to be more intuitive, but I can also change them back.